src/controllers/hid/hidcontroller.cpp

```diff
@@ -137,6 +137,70 @@ int HidController::close() {
     return 0;
 }
 
+void HidController::processInputReport(int& bytesRead, const int& currentBufferIndex) {
+    Trace process("HidController processInputReport");
+    unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+    unsigned char* pCurrentBuffer = m_pPollData[currentBufferIndex];
+    // Some controllers such as the Gemini GMX continuously send input reports even if it
+    // is identical to the previous send input report. If this loop processed all those redundant
+    // input report, it would be a big performance problem to run JS code for every  input report and
+    // would be unnecessary.
+    // This assumes that the redundant input report all use the same report ID. In practice we
+    // have not encountered any controllers that send redundant input report with different report
+    // IDs. If any such devices exist, this may be changed to use a separate buffer to store
+    // the last input report for each report ID.
+    if (bytesRead == m_iLastPollSize &&
+            memcmp(pCurrentBuffer, pPreviousBuffer, bytesRead) == 0) {
+        return;
+    }
+    m_iLastPollSize = bytesRead;
+    m_iPollingBufferIndex = currentBufferIndex;
+    auto incomingData = QByteArray::fromRawData(
+            reinterpret_cast<char*>(pCurrentBuffer), bytesRead);
+    receive(incomingData, mixxx::Time::elapsed());
+}
+
+QList<int> HidController::getInputReport(unsigned int reportID) {
+    Trace hidRead("HidController getInputReport");
+    int bytesRead;
+
+    // Cycle between buffers so the memcmp below does not require deep copying to another buffer.
+    const int currentBufferIndex = (m_iPollingBufferIndex + 1) % kNumBuffers;
```

Comment:
Why is m_iPollingBufferIndex left unchanged?

Reply:
This value was latched here, because it was used at the end of the function, after modifing the member variable inthe subfunction processInputReport.
I could remove this variable, by reordering the code, that the index is no longer needed after the call processInputReport.